app/controllers/spree/products_controller_decorator.rb

```diff
@@ -22,14 +23,9 @@ def index
   private
 
   def load_feed_products
-    items = []
     product_catalog = product_feed.try(:product_catalog)
-
-    Spree::Variant.where(id: product_catalog.item_ids).each do |variant|
-      items << Spree::ProductFeedService.new(variant)
-    end if product_catalog
-
-    @feed_products = items
+    item_ids = product_catalog ? product_catalog.item_ids : []
+    @feed_products = Spree::Variant.where(id: item_ids)
```

Comment:
I wonder if there is a way to cleanly combine these two lines to save yourself an unnecessary database call when item_ids is [], because that is always going to return []